server/src/signin-button.js

```diff
@@ -2,44 +2,69 @@ const React = require("react");
   constructor(props) {
     super(props);
     this.state = {
       displaySettings: props.isFxaAuthenticated,
+      showFxaOnboardingDialog: props.showFxaOnboardingDialog,
     };
   }
 
   static getDerivedStateFromProps(nextProps, prevState) {
-    return { displaySettings: nextProps.isFxaAuthenticated };
+    return { displaySettings: nextProps.isFxaAuthenticated};
+  }
+
+  componentDidUpdate(oldProps, oldState) {
+    if (oldProps.showFxaOnboardingDialog !== this.props.showFxaOnboardingDialog) {
+      this.setState({showFxaOnboardingDialog: this.props.showFxaOnboardingDialog});
+    }
   }
 
   render() {
     if (this.state.displaySettings) {
-      return <Localized id="buttonSettings" attrs={{title: true}}>
-          <a className="transparent nav-button icon-settings" tabIndex="0" href="/settings" title="Settings">
-            <img src={this.props.staticLink("/static/img/icon-settings.svg")} />
-          </a>
-        </Localized>;
+      return <div className="fxa-signin">
+          <Localized id="buttonSettings" attrs={{title: true}}>
+            <a className="transparent nav-button icon-settings" tabIndex="0" href="/settings" title="Settings">
+              <img src={this.props.staticLink("/static/img/icon-settings.svg")} />
+            </a>
+          </Localized>
+        </div>;
     }
 
     const logInURI = "/api/fxa-oauth/login/" + this.props.initialPage;
-    return <Localized id="buttonSignIn" attrs={{title: true}}>
-        <a className="transparent nav-button icon-settings" tabIndex="0" href={logInURI} title="SignIn"
-           onClick={this.clickHandler.bind(this)}>
-          <img src={this.props.staticLink("/static/img/icon-settings.svg")} />
-        </a>
-      </Localized>;
+    return <div className="fxa-signin">
+        <Localized id="buttonSignIn" attrs={{title: true}}>
+          <a className="transparent nav-button icon-settings" tabIndex="0" href={logInURI} title="SignIn"
+             onClick={this.clickHandler.bind(this)}>
+            <img src={this.props.staticLink("/static/img/icon-settings.svg")} />
+          </a>
+        </Localized>
+        <FxaOnboardingDialog logInURI={logInURI} display={this.state.showFxaOnboardingDialog}
+                             hideDialog={this.hideFxaOnboardingDialog.bind(this)} />
+      </div>;
   }
 
   clickHandler(event) {
     sendEvent("fxa-signin", this.props.initialPage, {useBeacon: true});
+    if (this.state.showFxaOnboardingDialog) {
+      this.hideFxaOnboardingDialog();
+    }
+  }
+
+  hideFxaOnboardingDialog() {
+    const showFxaOnboardingDialog = !this.state.showFxaOnboardingDialog;
+    this.setState({showFxaOnboardingDialog});
+    // set counter to max to stop showing promo again
+    localStorage.hasSeenOnboardingDialog = 3;
   }
 };
 
 exports.SignInButton.propTypes = {
   initialPage: PropTypes.string,
   isFxaAuthenticated: PropTypes.bool,
   staticLink: PropTypes.func,
+  showFxaOnboardingDialog: PropTypes.bool,
```

Comment:
I'd avoid using `showFoo` to mean the boolean state that decides whether we show Foo, because it seems like `showFoo` would be a show/hide method, not a boolean. This should be `shouldShowFxaOnboardingDialog`. It's long and clumsy, but is the least surprising name for what it does

Reply:
Updated to hasFxaOnboardingDialog